File: frontend/src/api/auth_reset.ts
import { apiClient } from '@/src/core/api/client';

export interface ResetPasswordRequest {
    email: string;
}

export interface ResetPasswordVerifyRequest {
    token: string;
}

export interface ResetPasswordConfirmRequest {
    token: string;
    new_password: string;
}

export const requestPasswordReset = async (data: ResetPasswordRequest): Promise<void> => {
    await apiClient.post('/api/auth/password-reset/request', data);
};

export const verifyResetToken = async (data: ResetPasswordVerifyRequest): Promise<void> => {
    await apiClient.post('/api/auth/password-reset/verify', data);
};

export const confirmPasswordReset = async (data: ResetPasswordConfirmRequest): Promise<void> => {
    await apiClient.post('/api/auth/password-reset/confirm', data);
};

File: backend/internal/server/dto/auth_reset.go
package dto

// ResetPasswordRequest provides ResetPasswordRequest related functionality.
type ResetPasswordRequest struct {
	Email string `json:"email"`
}

// ResetPasswordVerifyRequest provides ResetPasswordVerifyRequest related functionality.
type ResetPasswordVerifyRequest struct {
	Token string `json:"token"`
}

// ResetPasswordConfirmRequest provides ResetPasswordConfirmRequest related functionality.
type ResetPasswordConfirmRequest struct {
	Token       string `json:"token"`
	NewPassword string `json:"new_password"`
}
